drivers/i3c/i3c_common.c

```diff
@@ -1295,7 +1296,7 @@ int i3c_bus_deftgts(const struct device *dev)
 	}
 
 	/* Allocate memory for the struct with enough space for the targets */
-	deftgts = malloc(data_len);
+	deftgts = k_malloc(data_len);
```

Comment:
I agree with these changes... but my concern with this is that by default `CONFIG_HEAP_MEM_POOL_SIZE` is 0 and this will fail everytime, this probably should then be doing an adding of some bytes as followed by here (https://docs.zephyrproject.org/latest/kernel/memory_management/heap.html#defining-the-heap-memory-pool)

Maybe 32 bytes for default for I3C

Reply:
I don't see that option actually being used anywhere, and from the description alone I don't quite understand what it should do 😅 

https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr%20HEAP_MEM_POOL_ADD_SIZE_UHC_MCUX&type=code

Does it get used by linker scripts somehow? If you let me know what needs to be done, I'll update the PR no problem.

Reply:
They are collated and used yes https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/CMakeLists.txt#L135

Reply:
Never mind, I see plenty of `HEAP_MEM_POOL_ADD_SIZE_*`, I'll figure it out.